python/cudf/cudf/pylibcudf_tests/common/utils.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from typing import Optional
+
+import pyarrow as pa
+import pytest
+
+from cudf._lib import pylibcudf as plc
+
+
+def metadata_from_arrow_array(
+    pa_array: pa.Array,
+) -> Optional[plc.interop.ColumnMetadata]:
+    metadata = None
+    if pa.types.is_list(dtype := pa_array.type) or pa.types.is_struct(dtype):
+        metadata = plc.interop.ColumnMetadata(
+            "",
+            # libcudf does not store field names, so just match pyarrow's.
+            [
+                plc.interop.ColumnMetadata(pa_array.type.field(i).name)
+                for i in range(pa_array.type.num_fields)
+            ],
+        )
+    return metadata
+
+
+def assert_column_eq(plc_column: plc.Column, pa_array: pa.Array) -> None:
+    """Verify that the pylibcudf array and PyArrow array are equal."""
+    # Nested types require children metadata to be passed to the conversion function.
+    plc_pa = plc.interop.to_arrow(
+        plc_column, metadata=metadata_from_arrow_array(pa_array)
+    )
+
+    if isinstance(plc_pa, pa.ChunkedArray):
+        plc_pa = plc_pa.combine_chunks()
+    if isinstance(pa_array, pa.ChunkedArray):
+        pa_array = pa_array.combine_chunks()
+
+    assert plc_pa.equals(pa_array)
+
+
+def assert_table_eq(plc_table: plc.Table, pa_table: pa.Table) -> None:
```

Comment:
Is it important to make any assertions about the column names here? I figure no

Reply:
Yeah I've actually been discarding the names altogether. pyarrow requires names, but pylibcudf doesn't support them (since libcudf doesn't) so all the fixtures are creating pyarrow columns with empty names.